unit_test/docker_container_test.py

```diff
@@ -21,23 +25,178 @@
     DEFAULT_IMAGE = "quay.io/pypa/manylinux2014_ppc64le:2020-05-17-2f8ac3b"
 elif pm == "s390x":
     DEFAULT_IMAGE = "quay.io/pypa/manylinux2014_s390x:2020-05-17-2f8ac3b"
+else:
+    DEFAULT_IMAGE = ""
+
+
+temp_test_dir = None
+
+
+@atexit.register
+def _cleanup_tempdir():
```

Comment:
I wonder if this could be resolved more cleanly by using the pytest [tmp_path fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html). Did you try using that?

Reply:
I'll have to run tests. I the need to chown was an issue, but I don't remember the details.

Reply:
I think it's unlikely that the temp path fixture will work here. But I've been able to simplify the logic of the cleanup hook. 

The issue is that when you create a podman filesytem, it will create it with a fake UID of 0, which makes the system think you need to be root to remove it. This causes permission errors. I found a blog that documents the case: https://podman.io/blogs/2018/10/03/podman-remove-content-homedir.html

I was able to rewrite the logic using the "podman unshare" executable: 

```python
    global temp_test_dir
    if temp_test_dir is not None:
        print(f"CLEANUP temp_test_dir = {temp_test_dir!r}")  # type: ignore[unreachable]
        # When podman creates special directories, they cant be cleaned up
        # unless you fake a UID of 0. The package rootlesskit helps with that.
        import subprocess
        subprocess.call(['podman', 'unshare', 'rm', '-rf', temp_test_dir.name])
        try:
            temp_test_dir.cleanup()
        except Exception as ex:
            print(f"Issue cleaning up ex = {ex!r}")
    temp_test_dir = None
```

This deletes the temp directory without any permission issues. 

Is there a way to nicely turn this into a pytest hook? We also should only need to do this if the podman tests run (implying the podman executable is available), otherwise the existing tempdir cleanup mechanism should work.

Reply:
I modified the logic here. I've added another global called "using_podman", such that this only runs if the podman tests are being executed. I also had to move the atexit registration to be after the TemporaryDirectory was created, otherwise it would try to clean up the tempdir first.

I might be able to rewrite this as a pytest fixture with a module level scope. But I think it might depend on how `--run-podman` is implemented here, so I'm going to wait until those decisions are made to try messing with this. But I feel like I have a much better understanding of the issue I was working around now, and using `podman unshare` seems much cleaner than what I was doing.

Reply:
This looks much cleaner! The renames look very nice. I guess, my initial stab at using "cp" in `copy_out` had a mistake? Your refactor seems to use it, and the tests aren't breaking, so that's good.

The only thing I worry about is that the VFS configuration won't run through all of the container operations, but I've tested this branch on my use case and it doesn't break, so that's good enough for me. In general I'm happy with the state of the PR. I think it's ready to go